lib/streamlit/util.py

```diff
@@ -154,3 +155,33 @@ def calc_md5(s: str) -> str:
     h = hashlib.new("md5")
     h.update(s.encode("utf-8"))
     return h.hexdigest()
+
+
+def drop_key_query_params(
+    query_params: Dict[str, List[str]], keys_to_drop: List[str]
+) -> bool:
+    """Drop query_params (passed by reference) containing keys_to_drop: List[str] (case-insensitive) query params
+    Returns True if at least one param was dropped. False when there were no keys_to_drop found.
+    """
+    found = False
+    for param_name in copy.deepcopy(query_params).keys():
+        if param_name.lower() in keys_to_drop:
+            found = True
+            query_params.pop(param_name)
+    return found
```

Comment:
I'd avoid functions with side effects. There is a lot of things here that can go wrong.

I'd propose instead:

```
def drop_key_query_params(
    query_params: Dict[str, List[str]], keys_to_drop: List[str]
) -> Dict[str, List[str]]:
return {key:value for key, value in query_params.items() if key.lower() in keys_to_drop}
```

Alternatively, instead of dict, you can return a pair: dict, bool with bool answering the question weather the result is different than the input.

Reply:
Good call out, I've used lists comprehensions here, just as you suggested 👍